Sources/Sentry/SentryViewHierarchy.m

```diff
@@ -13,13 +13,20 @@ - (id)recursiveDescription;
 @implementation SentryViewHierarchy
 
 - (NSArray<NSString *> *)fetchViewHierarchy
+{
+    return [self fetchViewHierarchyPreventMoveToMainThread:NO];
+}
+
+- (NSArray<NSString *> *)fetchViewHierarchyPreventMoveToMainThread:(BOOL)preventMoveToMainThread
 {
     NSArray<UIWindow *> *windows = [SentryDependencyContainer.sharedInstance.application windows];
 
     NSMutableArray *result = [NSMutableArray arrayWithCapacity:[windows count]];
 
     [windows enumerateObjectsUsingBlock:^(UIWindow *window, NSUInteger idx, BOOL *stop) {
-        if ([NSThread isMainThread]) {
+        // In the case of a crash we can't dispatch work to be executed anymore,
+        // so we'll run this on the wrong thread.
+        if ([NSThread isMainThread] || preventMoveToMainThread) {
             [result addObject:[window recursiveDescription]];
```

Comment:
It used to be that using this private API (`recursiveDescription`) was known to get apps rejected from the App Store. Is that no longer the case?

Reply:
Yes this is still the case. 
I did not know that `recursiveDescription` is a private API, if that is the case we definitely need another solution.

Reply:
I don't think this is considered private? Yesterday I uploaded the sample app which includes this feature, and we didn't fail any automated checks.

I'm going to upload and actually submit for review another app today, let's see what happens. I'll include this PR into it.